packages/sdk/src/browser-wallets/unisat/index.ts

```diff
@@ -1,24 +1,133 @@
+import { getAddressFormat } from "../../addresses";
 import { OrditSDKError } from "../../errors";
-import type { BrowserWallet } from "../types";
+import { NETWORK_TO_UNISAT_NETWORK } from "./constants";
+import type { Psbt as PsbtType } from "bitcoinjs-lib";
+import type { Network } from "../../networks/types";
+import type { BrowserWalletSignResponse, WalletAddress } from "../types";
+import type { UnisatSignPSBTOptions } from "./types";
 
+/**
+ * Checks if the browser wallet extension is installed.
+ *
+ * @returns `true` if installed, `false` otherwise.
+ */
 function isInstalled() {
   if (typeof window === "undefined") {
     throw new OrditSDKError("Cannot call this function outside a browser.");
   }
   return typeof window.unisat !== "undefined";
 }
 
-async function getAddresses() {}
+/**
+ * Gets addresses from the browser wallet.
+ *
+ * @param network Network
+ * @returns An array of WalletAddress objects.
+ */
+async function getAddresses(
+  network: Network = "mainnet",
+): Promise<WalletAddress[]> {
+  if (!isInstalled()) {
+    throw new OrditSDKError("Unisat not installed.");
+  }
+
+  if (!network) {
+    throw new OrditSDKError("Invalid options provided.");
+  }
+
+  const targetNetwork = NETWORK_TO_UNISAT_NETWORK[network];
+  if (!targetNetwork) {
+    throw new OrditSDKError("Unsupported network.");
+  }
+
+  const connectedNetwork = await window.unisat.getNetwork();
+  if (connectedNetwork !== targetNetwork) {
+    await window.unisat.switchNetwork(targetNetwork);
+  }
+
+  const accounts = await window.unisat.requestAccounts();
+  const publicKey = await window.unisat.getPublicKey();
+
+  const address = accounts[0];
+  if (!address) {
+    return [];
+  }
+  const format = getAddressFormat(address, network);
+  return [
+    {
+      pub: publicKey,
+      address,
+      format,
+    },
+  ];
+}
+
+/**
+ * Signs a Partially Signed Bitcoin Transaction (PSBT).
+ * To learn more, visit https://github.com/bitcoin/bitcoin/blob/master/doc/psbt.md
+ *
+ * @param psbt Partially Signed Bitcoin Transaction
+ * @param options Options for signing
+ * @returns An object containing `base64` and `hex` if the transaction is not extracted, or `hex` if the transaction is extracted.
+ */
+async function signPsbt(
+  psbt: PsbtType,
+  options: UnisatSignPSBTOptions = {},
+): Promise<BrowserWalletSignResponse> {
+  if (!isInstalled()) {
+    throw new OrditSDKError("Unisat not installed.");
+  }
+
+  const { finalize = true, extractTx = true } = options;
+
+  const psbtHex = psbt.toHex();
+  const signedPsbtHex = await window.unisat.signPsbt(psbtHex, {
+    autoFinalized: finalize,
+  });
+  if (!signedPsbtHex) {
+    throw new OrditSDKError("Failed to sign psbt hex using Unisat.");
+  }
+
+  if (psbtHex === signedPsbtHex) {
+    throw new OrditSDKError("Psbt has already been signed.");
+  }
 
-async function signPsbt() {}
+  const Psbt = (await import("bitcoinjs-lib")).Psbt;
```

Comment:
I'm curious why it's a big no? I think the idea here is. If you're only using the other functionalities, you don't need to import bitcoinjs-lib (which is huge). Correct me if I'm wrong @Nanosync

Reply:
This approach would set a precedent and soon we will see inline imports everywhere in the code. I think this an additional overhead we will introduce that can be easily avoided. Best to follow the top-level imports.

Reply:
Is there a specific reason for no inline imports? Inline imports are being used here to ensure that we don't import the library immediately when importing this file. It's about 39.3kb (minified + gzipped) to import this lib (https://bundlephobia.com/package/bitcoinjs-lib@6.1.5)

I was checking the production build for this project using `vite preview` and `Psbt` didn't seem to be treeshaked even though I didn't call any signing related functions.

Reply:
> This approach would set a precedent and soon we will see inline imports everywhere in the code. I think this an additional overhead we will introduce that can be easily avoided. Best to follow the top-level imports.

That's the point. We should encourage it. :) The idea is to make this optimal not only for us, but also for the consumers. It will change a lot of things, but it will bring additional features that we never had.

Reply:
As mentioned above, this one change would set a precedent and void the convention of top-level imports that we have been following project-wide. We can separately discuss and improvise the code conventions that can eventually be implemented project-wide. 

I suggest we keep the PRs atomic and to the point, and not slide in changes to the coding convention that could otherwise be objected or debated on.

Reply:
Removed inline imports